src/v2/ion/uiSchemaLabelTransformer.js

```diff
@@ -0,0 +1,172 @@
+  'enroll-authenticator.security_question.sub_schema_local_credentials.1': 'oie.security.question.createQuestion.label',
+  'enroll-authenticator.security_question.credentials.answer': 'mfa.challenge.answer.placeholder',
+  'enroll-authenticator.security_question.credentials.question': 'oie.security.question.createQuestion.label',
+  'enroll-authenticator.security_question.credentials.questionKey': 'oie.security.question.questionKey.label',
+
+  'select-authenticator-authenticate.authenticator.email': 'oie.email',
+  'select-authenticator-authenticate.authenticator.password': 'oie.password',
+  'select-authenticator-authenticate.authenticator.phone': 'oie.phone',
+  'select-authenticator-authenticate.authenticator.security_key': 'oie.webauthn',
+  'select-authenticator-authenticate.authenticator.security_question': 'oie.security.question',
+
+  'challenge-authenticator.email.credentials.passcode': 'mfa.challenge.enterCode.placeholder',
+  'challenge-authenticator.password.credentials.passcode': 'mfa.challenge.password.placeholder',
+  'challenge-authenticator.phone.credentials.passcode': 'mfa.challenge.enterCode.placeholder',
+  'challenge-authenticator.security_question.credentials.answer': 'mfa.challenge.answer.placeholder',
+};
+
+const getI18nKey = (i18nPath) => {
+  let i18nKey;
+
+  // we can add mapping to `I18N_OVERRIDE_MAPPINGS` for all
+  // security question. It's just a bit tedious hence use following shortcut.
+  if (i18nPath.indexOf(SECURITY_QUESTION_PREFIX) === 0) {
+    const securityQuestionValue = i18nPath.replace(SECURITY_QUESTION_PREFIX, '');
+    i18nKey = `security.${securityQuestionValue}`;
+  }
+
+  if (I18N_OVERRIDE_MAPPINGS[i18nPath]) {
+    i18nKey = I18N_OVERRIDE_MAPPINGS[i18nPath];
+  }
+
+  if (i18nKey && !Bundles.login[i18nKey]) {
+    Logger.warn(`expect i18n key ${i18nKey} for ${i18nPath} but not found in 'login' bundle.`);
+  }
+
+  return i18nKey;
+};
+
+/**
+ * Find i18n value using {@code i18nPath} if it exists.
+ * Otherwise return {@code defaultValue}.
+ *
+ * @param {string} i18nPath
+ * @param {string} defaultValue
+ */
+const getI18NValue = (i18nPath, defaultValue) => {
+  const i18nKey = getI18nKey(i18nPath);
+  if (i18nKey) {
+    return loc(i18nKey, 'login');
+  } else {
+    return defaultValue;
+  }
+};
+
+const updateLabelForUiSchema = (remediation, uiSchema) => {
+  if (uiSchema.mutable === false) {
+    return;
+  }
+  Logger.info('i18n label transformer');
+  Logger.info('\t remediationName: ', remediation.name);
+  Logger.info('\t uiSchema: ', JSON.stringify(uiSchema));
+
+  const authenticatorType = remediation.relatesTo && remediation.relatesTo.value.type
+    ? `.${remediation.relatesTo.value.type}`
+    : '';
+  const i18nPrefix = `${remediation.name}${authenticatorType}.`;
+  const i18nPath = `${i18nPrefix}${uiSchema.name}`;
+
+  if (uiSchema.type === 'checkbox' && uiSchema.placeholder) {
```

Comment:
can we add some comments for these different checks ?

Reply:
I was thinking add comment but ended up with unit test which is probably bettor to elaborate use cases ?